File: Zerojudge-a302.cpp
// Author : ysh
// 2024/01/10 Wed 11:29:22
#include<bits/stdc++.h>
using namespace std;
const int R = 10007;
vector<int>mkf({1});
#define int long long
int fl(int x) {
    if(x == 0) return 1;
    if(mkf.size() > x) return mkf.at(x);
    mkf.push_back((x * (fl(x - 1))) % R);
    return mkf.at(x);
}
int ck(int a,int b) {
    if(b == 0) return 1;
    int tmp = ck(a,b >> 1);
    if(b & 1) return (((tmp * tmp) % R) * a) % R;
    return (tmp * tmp) % R;
}
int rv(int x) {
    return ck(x,R - 2);
}
int check(int a,int b) {
    return (((fl(a) * rv(fl(b))) % R) * rv(fl(a - b))) % R;
}
signed main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int a,b,c,d,e;cin>>a>>b>>c>>d>>e;
    cout<<(((check(c,d) * ck(a,d)) % R) * ck(b,e)) % R;
    return 0;
}

File: Zerojudge-a302.py
# Author : ysh
# 2024/01/10 Wed 11:45:55
import sys

input = sys.stdin.readline
print = sys.stdout.write

R = 10007
def fl(x) -> int:
    ans = 1
    for i in range(1,x + 1):
        ans = ans * i
    return ans

def ck(a,b) -> int:
    if b == 0: return 1;
    tmp = ck(a,b >> 1)
    if b & 1: return (((tmp * tmp) % R) * a) % R;
    return (tmp * tmp) % R

def rv(x) -> int:
    return ck(x,R- 2)

def check(a,b) -> int:
    return (((fl(a) * rv(fl(b))) % R) * rv(fl(a - b))) % R

a,b,c,d,e = [int(i) for i in input().split()]
print(str((((check(c,d) * ck(a,d)) % R) * ck(b,e)) % R))
quit()